icu4c/source/configure.ac

```diff
@@ -903,6 +903,14 @@ else
 fi
 AC_SUBST(U_HAVE_STRTOD_L)
 
+# Check if C++17 std::string_view is available.
```

Comment:
Do we need this?

Why not extend the conditions for U_CPLUSPLUS_VERSION in unicode/platform.h to detect C++17 (`__cplusplus >= 201703L`), and then make the test conditional on `#if U_CPLUSPLUS_VERSION >= 17`?

Reply:
In some environments (eg. current Apple Xcode), std::string_view is available even when a C++11 build is requested, so I thought it better to just run the std::string_view unit test whenever it's actually available instead of trying to test for conditions where it should be available.

Reply:
It adds a lot of "configure" machinery... but ok.